CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs

```diff
@@ -319,6 +327,135 @@ private void ExportListOfSolutionsToBeMerged(OrganizationServiceProxy servicePro
             }
         }
 
+        /// <summary>
+        /// Method gets deployment instance record
+        /// </summary>
+        /// <param name="serviceProxy">organization service proxy</param>
+        /// <param name="solutionFile">solution file info</param>        
+        private void GetDeploymentInstance(OrganizationServiceProxy serviceProxy, SolutionFileInfo solutionFile)
+        {
+            Entity sourceControl = solutionFile.Solution;
+            EntityCollection deploymentInstance = FetchDeplopymentInstance(serviceProxy, sourceControl.Id);
+
+            if (deploymentInstance.Entities.Count > 0)
+            {
+                foreach (Entity instance in deploymentInstance.Entities)
+                {
+                    ClientCredentials clientCredentials = new ClientCredentials();
+                    clientCredentials.UserName.UserName = instance.Attributes["syed_name"].ToString();
+                    clientCredentials.UserName.Password = instance.Attributes["syed_password"].ToString();
+                    OrganizationServiceProxy client = new OrganizationServiceProxy(new Uri(instance.Attributes["syed_instanceurl"].ToString()), null, clientCredentials, null);
+                    ImportSolution(client, solutionFile.SolutionFilePath ?? solutionFile.SolutionFilePathManaged);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Method import master solution to deployment instance
+        /// </summary>
+        /// <param name="serviceProxy">organization service proxy</param>
+        /// <param name="solutionImportPath">solution import path</param>
+        public static void ImportSolution(OrganizationServiceProxy serviceProxy, string solutionImportPath)
+        {
+            byte[] fileBytes = File.ReadAllBytes(solutionImportPath);
+
```

Comment:
you need to write in the log, starting deployment ot the organization + organizaitonserviceurl, same way completed deployment to the instance as well